include/hlsl++_vector_int.h

```diff
@@ -569,6 +569,16 @@ namespace hlslpp
 	hlslpp_inline int3& operator *= (int3& i1, const int3& i2) { i1 = i1 * i2; return i1; }
 	hlslpp_inline int4& operator *= (int4& i1, const int4& i2) { i1 = i1 * i2; return i1; }
 
+    hlslpp_inline bool any(const int1& i) { return 0 != i.x; }
```

Comment:
Thanks for the pull request @Benualdo. The problem with doing it this way and not having an hlslpp intrinsic is that on SIMD architectures you're going to swizzle 4 times, extract, do compares and AND them all when there are specific instructions like _mm_movemask that do this for you by returning a single int. There are float versions of these implemented for example in hlsl++_sse.h (look for _hlslpp_any4) which you can use for inspiration if you want. I'm not opposed to implementing these between today and tomorrow if you'd rather I do that :)

Reply:
If you would like to split the pull request I can approve the unimplemented comparison functions in hlsl++_scalar.h and the changes in hlsl++_sse.h where you add the comparison, basically all the changes except for the any/all.

Reply:
OK, I understand you that don't want to have 'any'/'all' implemented in an unoptimized way in hlslpp. I will make a new pull request with only the aforementioned changes as soon as possible today.